modulus/utils/generative/stochastic_sampler.py

```diff
@@ -0,0 +1,446 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import math
+from typing import Any, Callable, Optional
+
+import numpy as np
+import torch
+from torch import Tensor
+
+
+def image_batching(
+    input: Tensor,
+    img_shape_x: int,
+    img_shape_y: int,
+    patch_shape_x: int,
+    patch_shape_y: int,
+    batch_size: int,
+    overlap_pix: int,
+    boundary_pix: int,
+    input_interp: Optional[Tensor] = None,
+) -> Tensor:
+    """
+    Splits a full image into a batch of patched images.
+
+    This function takes a full image and splits it into patches, adding padding where necessary.
+    It can also concatenate additional interpolated data to each patch if provided.
+
+    Args:
+        input (Tensor): The input tensor representing the full image with shape
+                        (batch_size, channels, img_shape_x, img_shape_y).
+        img_shape_x (int): The width (x-dimension) of the original full image.
+        img_shape_y (int): The height (y-dimension) of the original full image.
+        patch_shape_x (int): The width (x-dimension) of each image patch.
+        patch_shape_y (int): The height (y-dimension) of each image patch.
+        batch_size (int): The original batch size before patching.
+        overlap_pix (int): The number of overlapping pixels between adjacent patches.
+        boundary_pix (int): The number of pixels to crop as a boundary from each patch.
+        input_interp (Optional[Tensor]): Optional additional data to concatenate to each patch
+                                         with shape (batch_size, interp_channels, patch_shape_x, patch_shape_y).
+
+    Returns:
+        Tensor: A tensor containing the image patches, with shape
+                (total_patches * batch_size, channels [+ interp_channels], patch_shape_x, patch_shape_y).
```

Comment:
Do we care that the doc format is not numpy?

Reply:
I care now that you have mentioned it :) 
Fixed it